saleor/graphql/product/filters.py

```diff
@@ -185,7 +185,13 @@ def _clean_product_attributes_date_time_range_filter_input(filter_value):
                     gte, datetime.datetime.min.time(), tzinfo=pytz.UTC
                 )
             filters["date_time__gte"] = gte
-    return matching_attributes.filter(**filters)
+    matching_queryset = matching_attribute_values.filter(**filters)
```

Comment:
Don't need to fetch all details as you're only using `pk` values. This could be reduce the amount of data to fetch  with `values_list`